src/parser/jobs/ast/modules/Sect/Sect.tsx

```diff
@@ -1,163 +1,90 @@
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
 import {Statistics} from 'parser/core/modules/Statistics'
 import Suggestions, {SEVERITY, Suggestion} from 'parser/core/modules/Suggestions'
 import React from 'react'
 import SectStatistic from './SectStatistic'
 
-const NO_SECT_ICON = 'https://xivapi.com/i/064000/064017.png'
+const noSectIcon = 'https://xivapi.com/i/064000/064017.png'
 
-const ASPECTED_ACTIONS = [
-	ACTIONS.ASPECTED_BENEFIC.id,
-	ACTIONS.ASPECTED_HELIOS.id,
-	ACTIONS.CELESTIAL_INTERSECTION.id,
-	ACTIONS.CELESTIAL_OPPOSITION.id,
+const SECT_STATUSES: StatusKey[] = [
+	'DIURNAL_SECT',
+	'NOCTURNAL_SECT',
 ]
 
-const DIURNAL_SECT_STATUSES = [
-	STATUSES.ASPECTED_BENEFIC.id,
-	STATUSES.ASPECTED_HELIOS.id,
-	STATUSES.DIURNAL_OPPOSITION.id,
-	STATUSES.NOCTURNAL_INTERSECTION.id,
-	STATUSES.NOCTURNAL_BALANCE.id,
+const DIURNAL_SECT_STATUSES: StatusKey[] = [
+	'ASPECTED_BENEFIC',
+	'ASPECTED_HELIOS',
+	'DIURNAL_OPPOSITION',
+	'NOCTURNAL_INTERSECTION',
+	'NOCTURNAL_BALANCE',
 ]
 
-const NOCTURNAL_SECT_STATUSES = [
-	STATUSES.NOCTURNAL_FIELD.id,
-	STATUSES.NOCTURNAL_OPPOSITION.id,
-	STATUSES.DIURNAL_INTERSECTION.id,
-	STATUSES.DIURNAL_BALANCE.id,
+const NOCTURNAL_SECT_STATUSES: StatusKey[] = [
+	'NOCTURNAL_FIELD',
+	'NOCTURNAL_OPPOSITION',
+	'DIURNAL_INTERSECTION',
+	'DIURNAL_BALANCE',
 ]
 
-const ACTION_STATUS_MAP = {
-	[ACTIONS.ASPECTED_BENEFIC.id]: [
-		STATUSES.ASPECTED_BENEFIC.id,
-		STATUSES.NOCTURNAL_FIELD.id,
-	],
-	[ACTIONS.ASPECTED_HELIOS.id]: [
-		STATUSES.ASPECTED_HELIOS.id,
-		STATUSES.NOCTURNAL_FIELD.id,
-	],
-	[ACTIONS.CELESTIAL_INTERSECTION.id]: [
-		STATUSES.DIURNAL_INTERSECTION.id,
-		STATUSES.NOCTURNAL_INTERSECTION.id,
-	],
-	[ACTIONS.CELESTIAL_OPPOSITION.id]: [
-		STATUSES.DIURNAL_OPPOSITION.id,
-		STATUSES.NOCTURNAL_OPPOSITION.id,
-	],
-}
-
-const SECT_ACTIONS = [
-	ACTIONS.DIURNAL_SECT.id,
-	ACTIONS.NOCTURNAL_SECT.id,
-]
-
-const SECT_STATUSES = [
-	STATUSES.DIURNAL_SECT.id,
-	STATUSES.NOCTURNAL_SECT.id,
-]
-
-const DIURNAL_SECT_BUFF_ABILITY = {
-	name: STATUSES.DIURNAL_SECT.name,
-	guid: STATUSES.DIURNAL_SECT.id,
-	type: 1,
-	abilityIcon: _.replace(_.replace(STATUSES.DIURNAL_SECT.icon, 'https://xivapi.com/i/', ''), '/', '-'),
-}
-
-const NOCTURNAL_SECT_BUFF_ABILITY = {
-	name: STATUSES.NOCTURNAL_SECT.name,
-	guid: STATUSES.NOCTURNAL_SECT.id,
-	type: 1,
-	abilityIcon: _.replace(_.replace(STATUSES.NOCTURNAL_SECT.icon, 'https://xivapi.com/i/', ''), '/', '-'),
-}
-
 // Determine sect by checking the result of an aspected spell/ability
-export default class Sect extends Module {
+export default class Sect extends Analyser {
 	static override handle = 'sect'
 
 	@dependency private statistics!: Statistics
-	@dependency private precastAction!: PrecastAction
 	@dependency private suggestions!: Suggestions
+	@dependency private data!: Data
 
 	private pullWithoutSect = false
-	private activeSectId: string | number | undefined = undefined
+	private activeSectId: Status['id'] = 0
```

Comment:
Why init to 0? NBD I guess if you're using it as a comparison, otherwise undefined is clearer, as if it's not set by diurnal/nocturnal then it's not defined anyway.

Reply:
updated to undefined